### PR TITLE
update xtreemfs and add nixos module

### PR DESCRIPTION
**Pull request by matejc**
- repackage xtreemfs
- nixos module

cc @offlinehacker and package maintainer @7c6f434c 

I was testing with this configuration (ssl and replication with two servers):

```
services = {
  xtreemfs = {
    enable = true;
    dir = {
      uuid = "0a7b017c-4e27-41f5-991a-dd80aa3b2106";
      extraConfig = ''
        ssl.enabled = true
        ssl.protocol = tlsv12
        ssl.service_creds.pw = <passphrase>
        ssl.service_creds.container = pkcs12
        ssl.service_creds = /etc/xos/xtreemfs/truststore/certs/dir.p12
        ssl.trusted_certs = /etc/xos/xtreemfs/truststore/certs/trusted.jks
        ssl.trusted_certs.pw = <jks_passphrase>
        ssl.trusted_certs.container = jks
      '';
      replication = {
        enable = true;
        extraConfig = ''
          babudb.repl.participant.0 = 192.168.0.11
          babudb.repl.participant.0.port = 35676
          babudb.repl.participant.1 = 192.168.0.10
          babudb.repl.participant.1.port = 35676
          babudb.repl.sync.n = 1
          babudb.ssl.enabled = true
          babudb.ssl.protocol = tlsv12
          babudb.ssl.service_creds = /etc/xos/xtreemfs/truststore/certs/osd.p12
          babudb.ssl.service_creds.pw = <passphrase>
          babudb.ssl.service_creds.container = pkcs12
          babudb.ssl.trusted_certs = /etc/xos/xtreemfs/truststore/certs/trusted.jks
          babudb.ssl.trusted_certs.pw = <jks_passphrase>
          babudb.ssl.trusted_certs.container = jks
          babudb.ssl.authenticationWithoutEncryption = false
        '';
      };
    };
    mrc = {
      uuid = "1c84471d-0337-4165-9572-fb865b4aee8f";
      extraConfig = ''
        osd_check_interval = 300
        no_atime = true
        local_clock_renewal = 0
        remote_time_sync = 30000
        authentication_provider = org.xtreemfs.common.auth.NullAuthProvider
        capability_secret = <secret>
        dir_service.host = 192.168.0.11
        dir_service.port = 32638
        dir_service.1.host = 192.168.0.10
        dir_service.1.port = 32638
        ssl.enabled = true
        ssl.protocol = tlsv12
        ssl.service_creds.pw = <passphrase>
        ssl.service_creds.container = pkcs12
        ssl.service_creds = /etc/xos/xtreemfs/truststore/certs/mrc.p12
        ssl.trusted_certs = /etc/xos/xtreemfs/truststore/certs/trusted.jks
        ssl.trusted_certs.pw = <jks_passphrase>
        ssl.trusted_certs.container = jks
      '';
      replication = {
        enable = true;
        extraConfig = ''
          babudb.repl.participant.0 = 192.168.0.11
          babudb.repl.participant.0.port = 35678
          babudb.repl.participant.1 = 192.168.0.10
          babudb.repl.participant.1.port = 35678
          babudb.repl.sync.n = 1
          babudb.ssl.enabled = true
          babudb.ssl.protocol = tlsv12
          babudb.ssl.service_creds = /etc/xos/xtreemfs/truststore/certs/osd.p12
          babudb.ssl.service_creds.pw = <passphrase>
          babudb.ssl.service_creds.container = pkcs12
          babudb.ssl.trusted_certs = /etc/xos/xtreemfs/truststore/certs/trusted.jks
          babudb.ssl.trusted_certs.pw = <jks_passphrase>
          babudb.ssl.trusted_certs.container = jks
          babudb.ssl.authenticationWithoutEncryption = false
        '';
      };
    };
    osd = {
        uuid = "d189172c-3f67-4d1f-87c6-d9fb1f826547";
        extraConfig = ''
          dir_service.host = 192.168.0.11
          dir_service.port = 32638
          dir_service.1.host = 192.168.0.10
          dir_service.1.port = 32638
          local_clock_renewal = 0
          remote_time_sync = 30000
          report_free_space = true
          capability_secret = <secret>
          ssl.enabled = true
          ssl.service_creds.pw = <passphrase>
          ssl.service_creds.container = pkcs12
          ssl.service_creds = /etc/xos/xtreemfs/truststore/certs/osd.p12
          ssl.trusted_certs = /etc/xos/xtreemfs/truststore/certs/trusted.jks
          ssl.trusted_certs.pw = <jks_passphrase>
          ssl.trusted_certs.container = jks
        '';
      };
    };
  };
};
```

### CODE REVIEW

File: nixos/modules/services/network-filesystems/xtreemfs.nix [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.xtreemfs = {
+
+      enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/network-filesystems/xtreemfs.nix]: Please use `mkEnableOption`.

**Comment by jagajaga** [comment]
Looks good!

**Comment by pSub** [comment]
I glanced over it and it looks good for me too. However there are currently conflicts.

**Comment by matejc** [comment]
Thanks for reviews, I will resolve those conflicts when I find time, today/tomorrow.
Just for the record: some of the services where crashing, I assumed that xtreemfs was configured wrong, but the firewall needed to be adjusted.. so this ports needed to be opened TCP: 32638 32636 32640 35676 35678 30638 30636 30640 and UDP: 32636 32638 32640.

**Pull request merged**